hypothesis-python/src/hypothesis/internal/conjecture/shrinker.py

```diff
@@ -121,11 +122,13 @@ def derived_value(fn):
     are calculated once, then cached until the shrink target changes, then
     recalculated the next time they are used."""
 
-    def accept(self):
+    def accept(instance):
         try:
-            return self.__derived_values[fn.__name__]
+            return instance._Shrinker__derived_values[fn.__name__]
```

Comment:
I think I'd prefer to leave this nested and just slap an ignore on it - explicit name mangling is usually a sign that you should be doing something else...

Reply:
Ok, fair enough! 👍

Reply:
Oh, also check `@staticmethod`.